### DISCUSSION TITLE
Feature: Simple Image Input

### DISCUSSION THREAD

**Discussion by imhappyfor** [facebook/lexical#3706] [Q&A]
I'm currently working on making an image input option in the toolbar, I usually wouldn't make a request like this but would it be possible for someone to provide a simple example of an icon image input button in the toolbar that inserts an image from the users computer to the text area? I've tried to use the playground code to make sense of how this is done but can't get anything working (in JS) and my code is currently in disarray. A codesandbox example would be greatly appreciated in either JS or TS(X). Cheers!

Sorry if this isn't the correct category label, it's the closest one I could find.

**Comment by evbo** [comment]
Not a great answer, but it appears drag & drop of images is being demonstrated here. Part of what was merged includes an updated sandbox, so if you pull this version you may have a code sample:
https://github.com/facebook/lexical/pull/3274

Also, there's a similar example where you first draw a modal dialog for the user to upload. Then, once on the server, give Lexical the url and it will display the image:
https://github.com/facebook/lexical/discussions/2835
